### PR TITLE
Remove `CreateAttachment::id` field

### PR DESCRIPTION
**Pull request by mkrasnitski**
For new attachments, the value of the `id` field corresponds to their order in the payload. Because these ids are sequential, we can simply calculate them when forming the multipart payload/serializing.

cc @kangalio

### CODE REVIEW

File: src/builder/create_attachment.rs
Comment (GnomedDev) [inline src/builder/create_attachment.rs:305]: Can this be rewritten to use `Serializer::collect_seq`?

**Review by GnomedDev — CHANGES_REQUESTED** [review]
Looks good, just a small improvement before I can approve

File: src/builder/create_attachment.rs
Comment (mkrasnitski) [reply src/builder/create_attachment.rs:305]: I don't think so, considering the types we're serializing are different between the two match arms.

Comment (GnomedDev) [reply src/builder/create_attachment.rs:305]: Ah yes, I don't know how I missed that, this would be possible if we had Either but probably not. I'll merge and look at it later if I get round to it.

**Review by GnomedDev — APPROVED** [review]

**Pull request merged**
